jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -73,6 +76,7 @@
     private boolean _dirAllowed=true;
     private boolean _redirectWelcome=false;
     private CompressedContentFormat[] _precompressedFormats=new CompressedContentFormat[0];
+    private final Map<String, List<String>> _preferredEncodingOrderCache = new ConcurrentHashMap<>();
```

Comment:
Using one of our Trie classes may be better than a ConcurrentHashMap.  Not certain, but worth a consideration...   hmmm not sure it is safe for concurrent updates...